src/test/java/uk/nhs/tis/trainee/notifications/service/HistoryServiceIntegrationTest.java

```diff
@@ -155,4 +164,43 @@ void shouldSortFoundNotificationsBySentAtWhenMultipleFound() {
     HistoryDto history3 = foundHistory.get(2);
     assertThat("Unexpected history sent at.", history3.sentAt(), is(before));
   }
+
+  @Test
+  void shouldNotRebuildMessageWhenNotificationNotFound() {
+    Optional<String> message = service.rebuildMessage(ObjectId.get().toString());
+
+    assertThat("Unexpected message.", message, is(Optional.empty()));
+  }
+
+  @ParameterizedTest
+  @MethodSource(
+      "uk.nhs.tis.trainee.notifications.MethodArgumentUtil#getEmailTemplateTypeAndVersions")
+  void shouldRebuildEmailMessageWhenNotificationFound(NotificationType notificationType,
```

Comment:
I'm not thrilled at this test, it does what it needs to but it feels awkward and brittle in the same way as some of the other tests that verify template content.
I'd like to look at the Thymeleaf testing library as a better way to test each template, but I'm at my boy scouting limits for this submission.